contracts/contracts/strategies/BridgedWOETHStrategy.sol

```diff
@@ -0,0 +1,301 @@
+        BaseStrategyConfig memory _stratConfig,
+        address _weth,
+        address _bridgedWOETH,
+        address _oethb
+    ) InitializableAbstractStrategy(_stratConfig) {
+        weth = IWETH9(_weth);
+        bridgedWOETH = IERC20(_bridgedWOETH);
+        oethb = IERC20(_oethb);
+    }
+
+    function initialize(uint128 _maxPriceDiffBps)
+        external
+        onlyGovernor
+        initializer
+    {
+        InitializableAbstractStrategy._initialize(
+            new address[](0), // No reward tokens
+            new address[](0), // No assets
+            new address[](0) // No pTokens
+        );
+
+        _setMaxPriceDiffBps(_maxPriceDiffBps);
+    }
+
+    /**
+     * @dev Sets the max price diff bps for the wOETH value appreciation
+     * @param _maxPriceDiffBps Bps value, 10k == 100%
+     */
+    function setMaxPriceDiffBps(uint128 _maxPriceDiffBps)
+        external
+        onlyGovernor
+    {
+        _setMaxPriceDiffBps(_maxPriceDiffBps);
+    }
+
+    /**
+     * @dev Sets the max price diff bps for the wOETH value appreciation
+     * @param _maxPriceDiffBps Bps value, 10k == 100%
+     */
+    function _setMaxPriceDiffBps(uint128 _maxPriceDiffBps) internal {
+        require(
+            _maxPriceDiffBps > 0 && _maxPriceDiffBps <= 10000,
+            "Invalid bps value"
+        );
+
+        emit MaxPriceDiffBpsUpdated(maxPriceDiffBps, _maxPriceDiffBps);
+
+        maxPriceDiffBps = _maxPriceDiffBps;
+    }
+
+    /**
+     * @dev Wrapper for _updateWOETHOraclePrice with nonReentrant flag
+     * @return The latest price of wOETH from Oracle
+     */
+    function updateWOETHOraclePrice() external nonReentrant returns (uint256) {
+        return _updateWOETHOraclePrice();
+    }
+
+    /**
+     * @dev Finds the value of bridged wOETH from the Oracle.
+     *      Ensures that it's within the bounds and reasonable.
+     *      And stores it.
+     *
+     *      NOTE: Intentionally not caching `Vault.priceProvider` here,
+     *      since doing so would mean that we also have to update this
+     *      strategy every time there's a change in oracle router.
+     *      Besides on L2, the gas is considerably cheaper than mainnet.
+     *
+     * @return Latest price from oracle
+     */
+    function _updateWOETHOraclePrice() internal returns (uint256) {
+        // WETH price per unit of bridged wOETH
+        uint256 oraclePrice = IOracle(IVault(vaultAddress).priceProvider())
+            .price(address(bridgedWOETH));
+
+        // 1 wOETH > 1 WETH, always
+        require(oraclePrice > 1 ether, "Invalid wOETH value");
+
+        uint128 oraclePrice128 = oraclePrice.toUint128();
+
+        // Do some checks
+        if (lastOraclePrice > 0) {
+            // Make sure the value only goes up
+            require(oraclePrice128 >= lastOraclePrice, "Negative wOETH yield");
+
+            // (lastOraclePrice * (1 + maxPriceDiffBps))
+            uint256 maxPrice = ((lastOraclePrice * (1e4 + maxPriceDiffBps)) /
```

Comment:
Having a consistent round up/down behaviour here sounds good. But I wonder if it's gonna cause any issues as it is in this place. 

As of now, that division truncates and doesn't round. We have `oraclePrice128 >= lastOraclePrice` just above. So, I can't really think of any case where this would fail. But I'm not sure if I'm missing anything here

Reply:
All good then 👍